app/src/main/java/com/example/android/oscarine/signin/SigninViewModel.kt

```diff
@@ -0,0 +1,61 @@
+package com.example.android.oscarine.signin
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.example.android.oscarine.SharedPreference
+import com.example.android.oscarine.network.OscarineApi
+import com.example.android.oscarine.network.models.login_user.LoginServerResponse
+import com.example.android.oscarine.network.models.login_user.LoginUser
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class SigninViewModel(private val sharedPreference: SharedPreference): ViewModel() {
+
+    private val _response = MutableLiveData<String>()
+    val response: LiveData<String>
+        get() = _response
```

Comment:
Instead of this, there should be a `messages` live data which you observe and show Toasts in the fragment